go-controller/pkg/ovn/egressip.go

```diff
@@ -2180,16 +2180,17 @@ func (e *egressIPController) addPodEgressIPAssignment(egressIPName string, statu
 	if err = e.deleteExternalGWPodSNAT(pod, podIPs, status); err != nil {
 		return err
 	}
-	if err = e.handleEgressReroutePolicy(podIPs, status, egressIPName, e.createEgressReroutePolicy); err != nil {
-		return fmt.Errorf("unable to create logical router policy, err: %v", err)
-	}
+
 	var ops []ovsdb.Operation
 	ops, err = createNATRuleOps(e.nbClient, podIPs, status, egressIPName)
 	if err != nil {
 		return fmt.Errorf("unable to create NAT rule for status: %v, err: %v", status, err)
 	}
 	_, err = libovsdbops.TransactAndCheck(e.nbClient, ops)
-	return err
+	if err != nil {
+		return fmt.Errorf("unable to apply NAT rule for status: %v, err: %v", status, err)
+	}
+	return e.handleEgressReroutePolicy(podIPs, status, egressIPName, e.createEgressReroutePolicy)
```

Comment:
why dont we just make this whole thing more efficient and do it as a single transaction? That way either the SNAT and the reroute policy get configured at the same time, or if an error happens neither of them are configured?

Reply:
Agreed, good idea.
I will work on refactoring the addition/deletion so they are always done as a single transaction.

Reply:
@trozet @tssurya done